src/linear_algebra/matrix.lean

```diff
@@ -1029,3 +1031,144 @@ lemma matrix.dot_product_eq_zero_iff {v : n → R} : (∀ w, matrix.dot_product
 ⟨λ h, matrix.dot_product_eq_zero v h, λ h w, h.symm ▸ zero_dot_product w⟩
 
 end
+
+namespace matrix
```

Comment:
Is there a reason to put this in the (already big) `matrix.lean` instead of `determinant.lean`?

Reply:
I tried to reorganize this, but these lemmas depend on matrix.det_reindex_self that is proved in matrix.lean. Trying to move `section reindexing` to determinant.lean runs into more dependencies on stuff in matrix.lean.